omf-bifrost/src/admin/mod.rs

```diff
@@ -0,0 +1,381 @@
+use anyhow::{Context, Result};
+use duckdb::Connection as DuckConnection;
+use log::info;
+use rusqlite::{Connection, params};
+use std::fs;
+use std::path::Path;
+
+mod config;
+
+pub use config::{
+    AccessMode, AdminConfig, HighwayType, load_admin_config, save_default_admin_config,
+};
+
+const CREATE_ADMINS_SQL: &str = "CREATE TABLE admins (
+                admin_level INTEGER NOT NULL,
+                iso_code TEXT,
+                parent_admin INTEGER,
+                name TEXT NOT NULL,
+                name_en TEXT,
+                drive_on_right INTEGER NULL,
+                allow_intersection_names INTEGER NULL,
+                default_language TEXT,
+                supported_languages TEXT,
+                division_id TEXT,
+                parent_division_id TEXT
+);";
+
+const ADD_ADMINS_GEOM_SQL: &str =
+    "SELECT AddGeometryColumn('admins', 'geom', 4326, 'MULTIPOLYGON', 2);";
+
+const CREATE_ADMIN_INDEX_SQL: &str = "CREATE INDEX IdxLevel ON admins ('admin_level');";
+const CREATE_ADMIN_DRIVE_INDEX_SQL: &str =
+    "CREATE INDEX IdxDriveOnRight ON admins ('drive_on_right');";
+const CREATE_ADMIN_INTERSECTION_INDEX_SQL: &str =
+    "CREATE INDEX IdxAllowIntersectionNames ON admins ('allow_intersection_names');";
+const CREATE_SPATIAL_INDEX_SQL: &str = "SELECT CreateSpatialIndex('admins', 'geom');";
+
+const CREATE_ADMIN_ACCESS_SQL: &str = "CREATE TABLE admin_access (
+            admin_id INTEGER NOT NULL,
+            iso_code TEXT,
+            trunk INTEGER DEFAULT NULL,
+            trunk_link INTEGER DEFAULT NULL,
+            track INTEGER DEFAULT NULL,
+            footway INTEGER DEFAULT NULL,
+            pedestrian INTEGER DEFAULT NULL,
+            bridleway INTEGER DEFAULT NULL,
+            cycleway INTEGER DEFAULT NULL,
+            path INTEGER DEFAULT NULL,
+            motorroad INTEGER DEFAULT NULL
+);";
+
+fn modes_to_bitmask(modes: &[AccessMode]) -> Option<i64> {
+    let bm = modes.iter().map(|m| m.bit()).fold(0, |acc, bit| acc | bit);
+    if bm == 0 { None } else { Some(bm) }
+}
+
+fn get_iso_code(admin_level: i64, country: &str, region: &Option<String>) -> Option<String> {
+    if admin_level == 2 {
```

Comment:
Should this be a constant? To get something like
```suggestion
    if admin_level == ADMIN_LEVEL_COUNTRY {
```